staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -907,16 +915,20 @@ func installAPI(s *GenericAPIServer, c *Config) {
 		routes.Index{}.Install(s.listedPathProvider, s.Handler.NonGoRestfulMux)
 	}
 	if c.EnableProfiling {
-		routes.Profiling{}.Install(s.Handler.NonGoRestfulMux)
-		if c.EnableContentionProfiling {
-			goruntime.SetBlockProfileRate(1)
+		if s.SocketProfiling != nil {
+			s.SocketProfiling.Install()
+		} else {
+			routes.Profiling{}.Install(s.Handler.NonGoRestfulMux)
+			if c.EnableContentionProfiling {
+				goruntime.SetBlockProfileRate(1)
+			}
+			// so far, only logging related endpoints are considered valid to add for these debug flags.
+			routes.DebugFlags{}.Install(s.Handler.NonGoRestfulMux, "v", routes.StringFlagPutHandler(logs.GlogSetter))
```

Comment:
exposing the logs verbosity override on the debug socket seems pretty useful.

Reply:
Agreed